### PR TITLE
feat: use viper in price-feeder config

### PR DESCRIPTION
**Pull request by rbajollari**
## Description

closes: https://github.com/umee-network/umee/issues/1134

Replaced existing toml config package with viper to read a toml config in the price-feeder and allow environment variable configuring as well. Added an extra unit test in config_test.go to test environment variables taking precedence over config file. 

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] added appropriate labels to the PR
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/umee-network/umee/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: price-feeder/config/config.go [hunk 1]
```diff
@@ -60,92 +59,92 @@ var (
...
+		TMRPCEndpoint string `mapstructure:"tmrpc_endpoint" validate:"required"`
+		GRPCEndpoint  string `mapstructure:"grpc_endpoint" validate:"required"`
+		RPCTimeout    string `mapstructure:"rpc_timeout" validate:"required"`
 	}
 
 	// Telemetry defines the configuration options for application telemetry.
 	Telemetry struct {
 		// Prefixed with keys to separate services
-		ServiceName string `toml:"service_name" mapstructure:"service-name"`
+		ServiceName string `mapstructure:"service_name" mapstructure:"service-name"`
```
Comment (adamewozniak) [inline price-feeder/config/config.go:122]: Let's see if we can use the cosmos sdk's telemetry config struct instead. Ref: https://github.com/umee-network/umee/pull/1121/files#r918038413

File: price-feeder/config/config.go [hunk 2]
```diff
@@ -186,12 +185,19 @@ func ParseConfig(configPath string) (Config, error) {
...
+	viper.AutomaticEnv()
+	viper.SetConfigFile(configPath)
+	viper.SetConfigType("toml")
+
+	if err := viper.ReadInConfig(); err != nil {
+		if _, ok := err.(viper.ConfigFileNotFoundError); ok {
+			return cfg, fmt.Errorf("failed to read config because file was not found: %w", err)
+		} else {
+			return cfg, fmt.Errorf("failed to read config: %w", err)
+		}
```
Comment (adamewozniak) [inline price-feeder/config/config.go:197]: The underlying error will have the same information, might as well keep the code simple:
```suggestion
			return cfg, fmt.Errorf("failed to read config: %w", err)
```

File: price-feeder/config/config_test.go [hunk 3]
```diff
@@ -131,7 +131,7 @@ func TestValidate(t *testing.T) {
 }
 
 func TestParseConfig_Valid(t *testing.T) {
-	tmpFile, err := ioutil.TempFile("", "price-feeder.toml")
+	tmpFile, err := ioutil.TempFile("", "price-feeder*.toml")
```
Comment (adamewozniak) [inline price-feeder/config/config_test.go:134]: Why was this changed to `price-feeder*.toml`?

File: price-feeder/config/config_test.go [hunk 4]
```diff
@@ -527,3 +527,92 @@ global_labels = [["chain-id", "umee-local-testnet"]]
 	_, err = config.ParseConfig(tmpFile.Name())
 	require.Error(t, err)
 }
+
+func TestParseConfig_Env_Vars(t *testing.T) {
```
Comment (adamewozniak) [inline price-feeder/config/config_test.go:531]: Nice test!

**Review by adamewozniak — COMMENTED** [review]
Just a couple suggestions but good job 🥳 also please add a changelog entry for the price feeder

File: price-feeder/config/config_test.go (hunk 3 above)
Comment (rbajollari) [reply price-feeder/config/config_test.go:134]: ioutil.TempFile was appending a random string to the file name (ex: price-feeder.toml123457). This was making viper throw an error for an invalid file extension. The * moves the random string to where its set (ex: price-feeder123457.toml in this case) and away from the file extension.

File: price-feeder/CHANGELOG.md [hunk 5]
```diff
@@ -69,6 +69,7 @@ Ref: https://keepachangelog.com/en/1.0.0/
 - [#1170](https://github.com/umee-network/umee/pull/1170) Restrict price feeder quotes to USD, USDT, USDC, ETH, DAI, and BTC.
 - [#1175](https://github.com/umee-network/umee/pull/1175) Add ProviderName type to facilitate the reading of maps.
 - [#1215](https://github.com/umee-network/umee/pull/1215) Moved ProviderName to Name in provider package.
+- [#1274](https://github.com/umee-network/umee/pull/1274) Use viper in price-feeder config
```
Comment (adamewozniak) [inline price-feeder/CHANGELOG.md:72]:
```suggestion
- [#1274](https://github.com/umee-network/umee/pull/1274) Add option to set config by env variables.
```

**Review by adamewozniak — APPROVED** [review]
one nitpick but lgtm! make sure tests pass & then let's merge this on in, great work

**Comment by adamewozniak** [comment]
Just need these to get signed and we'll be gouda 🥳
